plugins/inputs/prometheus/kubernetes.go

```diff
@@ -98,15 +99,44 @@ func (p *Prometheus) watch(ctx context.Context, client *k8s.Client) error {
 
 			switch eventType {
 			case k8s.EventAdded:
-				registerPod(pod, p)
-			case k8s.EventDeleted:
-				unregisterPod(pod, p)
+				if podReady(pod.Status.GetContainerStatuses()) {
+					registerPod(pod, p)
+				}
 			case k8s.EventModified:
+				if pod.Metadata.GetDeletionTimestamp() != nil && podConditionReady(pod.Status.GetConditions()) {
+					unregisterPod(pod, p)
+				} else if podReady(pod.Status.GetContainerStatuses()) && podConditionReady(pod.Status.GetConditions()) {
+					registerPod(pod, p)
```

Comment:
What if the pod was already registered in EventAdded?  Since `p.kubernetesPods` is a list won't we add it twice?  I think this might be simpler if we change it to a map.

Reply:
I guess I didn't try a pod restart, which would trigger a modify event on an already registered pod.. I'll test that and see.

Reply:
This particular case is handled now. Though it would only be triggered if kubernetes didn't initiate the deletion of the old pod (`docker kill mypod...`).